Use the correct ShadowPalette for d2k paradrops

abcdefg30 opened the pull request:
Followup from #8719.
Fixes this crash when paradroping units on bleed:

```
Dune 2000 Mod at Version {DEV_VERSION}
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.34209
Exception of type `System.InvalidOperationException`: Palette `player` does not exist
   at OpenRA.Graphics.HardwarePalette.GetPalette(String name) in \OpenRA\OpenRA.Game\Graphics\HardwarePalette.cs:Line 46.
   at OpenRA.Graphics.WorldRenderer.CreatePaletteReference(String name) in \OpenRA\OpenRA.Game\Graphics\WorldRenderer.cs:Line 65.
   at OpenRA.Exts.GetOrAdd[K,V](Dictionary`2 d, K k, Func`2 createFn) in \OpenRA\OpenRA.Game\Exts.cs:Line 129.
   at OpenRA.Graphics.WorldRenderer.Palette(String name) in \OpenRA\OpenRA.Game\Graphics\WorldRenderer.cs:Line 69.
   at OpenRA.Mods.Common.Traits.WithParachute.<Render>d__e.MoveNext()
   at OpenRA.Actor.<Renderables>d__d.MoveNext() in \OpenRA\OpenRA.Game\Actor.cs:Line 156.
   at System.Linq.Enumerable.<SelectManyIterator>d__14`2.MoveNext()
   at System.Linq.Enumerable.<ConcatIterator>d__71`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.OrderedEnumerable`1.<GetEnumerator>d__0.MoveNext()
   at System.Linq.Enumerable.<ConcatIterator>d__71`1.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at OpenRA.Graphics.WorldRenderer.GenerateRenderables() in \OpenRA\OpenRA.Game\Graphics\WorldRenderer.cs:Line 117.
   at OpenRA.Graphics.WorldRenderer.Draw() in \OpenRA\OpenRA.Game\Graphics\WorldRenderer.cs:Line 134.
   at OpenRA.Game.RenderTick() in \OpenRA\OpenRA.Game\Game.cs:Line 525.
   at OpenRA.Game.Loop() in \OpenRA\OpenRA.Game\Game.cs:Line 653.
   at OpenRA.Game.Run() in \OpenRA\OpenRA.Game\Game.cs:Line 671.
   at OpenRA.Program.Run(String[] args) in \OpenRA\OpenRA.Game\Support\Program.cs:Line 111.
   at OpenRA.Program.Main(String[] args) in \OpenRA\OpenRA.Game\Support\Program.cs:Line 39.
```

pchote commented:
Can we please fix the default instead?  Defaulting `ShadowPalette` to `player` doesn't make a lot of sense.

abcdefg30 committed:
Replace the default value of ShadowPalette with "shadow".

abcdefg30 committed:
Fix WithParachute on ^Vehicle

abcdefg30 committed:
Fix bogus paradropping in survival01

abcdefg30 commented:
Updated.

pchote commented on mods/ra/rules/defaults.yaml:
```diff
@@ -128,7 +128,12 @@
 	WithParachute:
 		UpgradeTypes: parachute
 		UpgradeMinEnabledLevel: 1
+		ShadowPalette: player
```
Why is this necessary?

abcdefg30 replied:
Doesn't seem like we need this.

obrakmann commented:
The scripted infantry paradrops in at least the beginning of Survival01 and Intervention don't seem to work anymore, either. (not saying that it's this PR's fault, but that it needs to be fixed.)

obrakmann commented:
I figured it out. The scripted parachuting doesn't work because in #8339, the parachute was changed from an effect to an activity. However, idle triggers in the scripts cause the units to already have an activity, so the `Parachute` activity is queued and would only run once the initial activity (usually a `Hunt` or `AttackMove`) is finished.

abcdefg30 commented:
Updated and removed `ShadowPalette: player`.

obrakmann committed:
Fix scripted paradrops

A recent PR changed ParaDrop from an Effect to an Activity. Idle triggers in the mission scripts however have already queued activities for the paradropped units, so the ParaDrop activity would only run once the first activity had finished (which could never happen).

obrakmann commented:
Fixed the scripted paradrops in https://github.com/abcdefg30/OpenRA/pull/6.

abcdefg30 commented:
Merged.

obrakmann commented:
Both d2k and ra paradrops (both infantry and vehicles) work again, :+1:

Mailaender commented:
Paratroopers work in skirmish and Survival 02 as well as Dune 2000. :white_check_mark:
